src/middlewared/middlewared/plugins/chart_releases_linux/portals.py

```diff
@@ -73,6 +73,16 @@ def tag_func(key):
 
         return cleaned_portals
 
+    @private
+    def get_ix_chart_portal(self, release_data, node_ip):
+        portal_config = release_data['config'].get('portalDetails')
+        if not portal_config or not release_data['config'].get('enableUIPortal'):
+            return {}
+        host = node_ip if portal_config['useNodeIP'] else portal_config['host']
+        return {
+            portal_config['portalName']: [f'{portal_config["protocol"]}://{host}:{portal_config["port"]}']
+        }
```

Comment:
Preferably this gets writhen with a nested dicts to follow the standing practice of allowing multiple portals to be defined.

Reply:
`ix-chart` is a simple one docker image deployment, i don't see any reason for making this confusing to an average user to allow multiple portals. Do you ?

Reply:
I'm not saying you should expose multiple portals, I'm saying to at least allow for it in the backend so other catalogs can actually use it. Considering support for multiple portals already exists.

Also:
This fix would allow us to significantly simplify we expose our own portals (not user defined) as well, as with this we can just deploy the portals from values.yaml directly.

Reply:
I don't follow, other catalogs/apps can already specify as many portals as they want ? This is just specific to official catalogs `ix-chart` app ?

Reply:
No we cannot define portals from values.yaml.
At least not enable or disable them dynamically.

We have had to implement very hacky configmap code to even support dynamic portals at all.
---
Also:
Having "iX Only" features added is a bad practice to begin with, I don't understand what the issue is... When A downstream tells you "please just make this function accessable for everyone not just yourself as we want to use it as well", what is the issue against doing so?

Reply:
@Ornias1993 if a chart dev is writing an application, can you please tell me how the portal is going to be dynamic ? i.e i write an application for nginx for example, i know it will always run this `X` path because i wrote the app, how is it really dynamic ?
About the user feedback, users can specify protocol/port already, so how does this help ?

Reply:
```
Also:
Having "iX Only" features added is a bad practice to begin with, I don't understand what the issue is... When A downstream tells you "please just make this function accessable for everyone not just yourself as we want to use it as well", what is the issue against doing so?
```
Problem is i am not even sure how/why the downstream is requesting this or what the use case is even in general ? These changes are just for apps which are deployed by user themselves via `ix-chart`. It does not make sense for any other application.
So it would be nice if you can clarify what you think we should have and why ?

Reply:
> @Ornias1993 if a chart dev is writing an application, can you please tell me how the portal is going to be dynamic ? i.e i write an application for nginx for example, i know it will always run this X path because i wrote the app, how is it really dynamic ?
About the user feedback, users can specify protocol/port already, so how does this help ?

It's called "different points of entry", remember we support native kubernetes ingress.
You don't know which entrypoint a user is going to configure and currently we have to hack at it to apply this because questions.yaml doesn't accept templating and this would solve that issue instantly.

Byond that: We also have had to deploy apps that have per-install requirements for custom URL entrypoints with tokens. 

> Problem is i am not even sure how/why the downstream is requesting this or what the use case is even in general ?

The problem is the need to ask, because it's kinda obvious that entrypoints are not always the static.

> These changes are just for apps which are deployed by user themselves via ix-chart. It does not make sense for any other application.

As I always like to repeat: Assumptions are the mother of all fuckups. You just "assume" it doesn't make sense, based on the limited featureset of the iX Helm-Charts/Apps. Which is an inherently flawed way of dealing with these things.

But even assuming it only makes sense with "user customisable deployments", you guys already know for ages we have TrueCharts Custom App as an iX-Chart alternative.

So this whole "explain... explain... explain..." looks somewhat as a charade to sugarcoat a wish to have ix-only "specialsauces" added.


---

It's kinda annoying to have to spend time to explain quite standard kubernetes functionality/requirements/behavior/design-choices.

Reply:
TLDR:
Our (and most other!) Helm-Charts are a lot more complex than anything in the iX-Catalog. such as:
- Supporting multiple loadbalancers, nodeport and ingress (some of which at the same time!)
- Having Apps with very specific per-install dependend url requirements

Reply:
Thank you for clarifying, but this requires more thought wrt implementation as it should be much more generic then what is being offered here. Can you please make a ticket at jira how you propose the implementation should look like and we can take it from there ? Thanks!

Reply:
From the values.yaml side of things, it would just mean to not have that freaking ix-chart-only flag.
Really, I need to explain how values.yaml doesn't discriminate between ix-chart and every other helm-chart in existence?

If you can parse those in ix-chart, you can in every other chart.